### PR TITLE
Add RTL support to scrolling

### LINKED ISSUE #11764: SliverPadding RTL

### LINKED ISSUE #11836: PageView RTL

### PR DESCRIPTION
**Pull request by abarth**
This patch includes:

 * SliverPadding
 * ScrollView
 * PageView

Fixes #11764
Fixes #11836

### CODE REVIEW

File: packages/flutter/lib/src/rendering/sliver_padding.dart [hunk 1]
```diff
@@ -29,24 +29,47 @@ class RenderSliverPadding extends RenderSliver with RenderObjectWithChildMixin<R
   ///
   /// The [padding] argument must not be null and must have non-negative insets.
   RenderSliverPadding({
-    @required EdgeInsets padding,
+    @required EdgeInsetsGeometry padding,
+    TextDirection textDirection,
     RenderSliver child,
   }) : assert(padding != null),
-       assert(padding.isNonNegative),
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/sliver_padding.dart]: I believe isNonNegative is actually on EdgeInsetsGeometry now. I may have failed to update Padding when I changed that.

**Comment by Hixie** [comment]
LGTM

These patches are very satisfying.

**Comment by abarth** [comment]
> These patches are very satisfying.

Thanks.  This patch felt like our plan worked since we did all the crazy hard rtl scrolling work as part of the sliver project.

File: packages/flutter/test/widgets/list_view_horizontal_test.dart [hunk 2]
```diff
@@ -5,31 +5,35 @@
 import 'package:flutter_test/flutter_test.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
```
Comment (Hixie) [inline packages/flutter/test/widgets/list_view_horizontal_test.dart]: this should import foundation

Comment (Hixie) [reply packages/flutter/test/widgets/list_view_horizontal_test.dart]: (also, sort these alphabetically)

File: packages/flutter/lib/src/rendering/sliver_padding.dart (hunk 1 above)
Comment (abarth) [reply packages/flutter/lib/src/rendering/sliver_padding.dart]: Looks like padding has it.  Not sure why I removed this.

File: packages/flutter/test/widgets/list_view_horizontal_test.dart (hunk 2 above)
Comment (abarth) [reply packages/flutter/test/widgets/list_view_horizontal_test.dart]: Done.

**Comment by Hixie** [comment]
It's hard to be sure but I suspect this patch regressed stock_build_iteration. (So did https://github.com/flutter/flutter/pull/11709.)

**Pull request merged**